src/main/java/org/polydev/gaea/population/PopulationManager.java

```diff
@@ -39,11 +42,22 @@ public void populate(@NotNull World world, @NotNull Random random, @NotNull Chun
             needsPop.add(new ChunkCoordinate(chunk));
             int x = chunk.getX();
             int z = chunk.getZ();
+            GlueList<CompletableFuture<Chunk>> chunks = new GlueList<>();
             if(main.isEnabled()) {
+                int i = 0;
```

Comment:
We don't need all these chunks. Would be more performant to get only the chunks we actually need.